File: app/src/main/java/com/example/uber/helper/Local.java
package com.example.uber.helper;

import android.location.Location;

import com.google.android.gms.maps.model.LatLng;

import java.text.DecimalFormat;

public class Local {

    public static float cacularDistancia(LatLng latLngInicial, LatLng latLngFinal) {

        Location localInicial = new Location("Local inicial");
        localInicial.setLatitude(latLngInicial.latitude);
        localInicial.setLongitude(latLngInicial.longitude);

        Location localFinal = new Location("Local final");
        localFinal.setLatitude(latLngFinal.latitude);
        localFinal.setLongitude(latLngFinal.longitude);

        // Calcular distancia - Resultado em metros
        // dividir por 1000 para converter em Km
        float distancia = localInicial.distanceTo(localFinal) / 1000;

        return distancia;

    }

    public static String formatarDistancia(float distancia) {

        String distanciaFormatada;

        if (distancia < 1) {
            distancia = distancia * 1000; // em metros
            distanciaFormatada = Math.round(distancia) + " m ";
        } else {
            DecimalFormat decimal = new DecimalFormat("0.0");
            distanciaFormatada = decimal.format(distancia) + " Km ";
        }

        return distanciaFormatada;

    }

}

File: app/src/main/java/com/example/uber/helper/Local.kt
package com.example.uber.helper

import android.location.Location
import com.google.android.gms.maps.model.LatLng
import java.text.DecimalFormat

class Local {

    companion object {

        @JvmStatic
        fun cacularDistancia(latLngInicial: LatLng, latLngFinal: LatLng): Float {

            val localInicial = Location("Local inicial")
            localInicial.latitude = latLngInicial.latitude
            localInicial.longitude = latLngInicial.longitude

            val localFinal = Location("Local final")
            localFinal.latitude = latLngFinal.latitude
            localFinal.longitude = latLngFinal.longitude

            val distancia: Float
            distancia = localInicial.distanceTo(localFinal) / 1000

            return distancia

        }

        @JvmStatic
        fun formatarDistancia(distancia: Float): String {

            val distanciaFormatada: String
            val novaDistancia: Float
            val decimal = DecimalFormat("0.0")

            if ( distancia < 1 ) {

                novaDistancia = distancia * 1000
                distanciaFormatada = Math.round(novaDistancia).toString() + " m"

            } else {

                distanciaFormatada = decimal.format(distancia.toDouble()) + " Km"

            }

            return distanciaFormatada

        }

    }

}